src/CommandLine.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include "Filesystem.hpp"
+
+#include <cxxopts/include/cxxopts.hpp>
+
+#include <optional>
+#include <string>
+
+namespace tsm {
+
+/***********************************************************************************/
+class CommandLine final {
+
+public:
+    struct [[nodiscard]] Options {
+        std::string InputDir;
+        std::string DatasetName;
+        std::string OutputDir;
+        std::string RegexPattern;
+        fs::path FileListPath;
+        std::string RegexEngine;
+        std::vector<fs::path> STDINfileList;
+        bool DryRun{ false };
+        bool KeepIndexFile{ false };
+        bool RegenIndices{ false };
+        bool Forecast{ false };
+        bool Historical{ false };
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [struct member 'Options::Historical' is never used.](https://app.codacy.com/manual/htmlboss/netcdf-timestamp-mapper/pullRequest?prid=5068050)